lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -561,7 +561,7 @@ protected Schema createSchema(Type type, Map<String,Schema> names) {
         return Schema.create(Schema.Type.BYTES);
       if (Collection.class.isAssignableFrom(c))              // array
         throw new AvroRuntimeException("Can't find element type of Collection");
-      for (Conversion<?> conversion : conversions.values()) {  // logical type
+      for (Conversion<?> conversion : getAllConversions()) { // logical type
```

Comment:
shouldn't we be able to use the existing look ups to grab the conversion that corresponds to the class?